sycl-jit/jit-compiler/lib/KernelFusion.cpp

```diff
@@ -272,16 +308,43 @@ compileSYCL(InMemoryFile SourceFile, View<InMemoryFile> IncludeFiles,
                                       Verbose);
   }
 
-  auto ModuleOrErr =
-      compileDeviceCode(SourceFile, IncludeFiles, UserArgList, BuildLog);
-  if (!ModuleOrErr) {
-    return errorTo<RTCResult>(ModuleOrErr.takeError(),
-                              "Device compilation failed");
+  std::unique_ptr<llvm::Module> Module;
+
+  if (CachedIR.size() > 0) {
+    llvm::StringRef IRStr{CachedIR.begin(), CachedIR.size()};
+    std::unique_ptr<llvm::MemoryBuffer> IRBuf =
+        llvm::MemoryBuffer::getMemBuffer(IRStr, /*BufferName=*/"",
+                                         /*RequiresNullTerminator=*/false);
+    auto ModuleOrError = llvm::parseBitcodeFile(*IRBuf, Context);
```

Comment:
Does this fail gracefully in all cases or do we need to catch exceptions in some circumstances?

Reply:
Okay, as long as all LLVM-specific issues are handled, that should be fine.